src/stories/3-toggle.stories.tsx

```diff
@@ -1,30 +1,64 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { action } from '@storybook/addon-actions'
 import { Box } from 'theme-ui'
 import { Toggle } from '../components/Toggle'
 
 export default { title: 'Toggle' }
 
-export const AllState = () => (
-  <Box m={2}>
+const ToggleWithState = ({ label, ...rest }: { label: string }) => {
+  const [checked, setChecked] = useState(false)
+  return (
     <Box>
-      <Toggle checked={true} disabled={false} onChange={action('clicked')}>
-        Enabled checked
+      <Toggle
+        checked={checked}
+        disabled={false}
+        onChange={(c: boolean) => setChecked(c)}
+        {...rest}
+      >
+        {label}
       </Toggle>
     </Box>
-    <Box>
-      <Toggle checked={false} disabled={false} onChange={action('clicked')}>
-        Enabled unchecked
-      </Toggle>
-    </Box>
-    <Box>
-      <Toggle checked={true} disabled={true} onChange={action('clicked')}>
-        Disabled checked
+  )
+}
+
+export const AllState = () => {
+  return (
+    <Box m={2}>
+      <ToggleWithState label="Toggle checkbox" />
+      <ToggleWithState
+        label="Toggle with inline label"
+        sx={{
+          display: 'flex',
+          alignItems: 'center',
+          '* + *': {
+            ml: 1,
+          },
+        }}
+      />
+
+      <Box>
+        <Toggle
+          checked={true}
+          disabled={true}
+          onChange={action('This is not going to happen')}
+        >
+          Disabled checked
+        </Toggle>
+      </Box>
+
+      <Toggle
+        checked={false}
+        disabled={true}
+        onChange={action('This is not going to happen')}
+      >
+        Disabled unchecked
       </Toggle>
     </Box>
+  )
+}
 
-    <Toggle checked={false} disabled={true} onChange={action('clicked')}>
-      Disabled unchecked
-    </Toggle>
-  </Box>
-)
+AllState.parameters = {
```

Comment:
this was an attempt to have documentation but it doesn't work we might need to look at how to implement this bit I got it from here https://github.com/storybookjs/storybook/blob/next/examples/official-storybook/stories/demo/button.stories.js#L26